### PR TITLE
[GPT2] Capture an error on unicode support bug

### PR DESCRIPTION
**Pull request by texasmichelle**
This is a temporary patch to prevent crashes caused by a bug in `BytePairEncoder` support for multibyte unicode scalars, described in #385

### CODE REVIEW

File: Models/Text/GPT2/GPT2.swift [hunk 1]
```diff
@@ -155,15 +154,13 @@ public class GPT2 {
...
-          throw GPT2Error.invalidSeed(seed: seed)
-        }
+        let id = Int32(seed[0][0])!
         if id == Int32(endOfTextId) {
             // Replace with newline.
             return "\r\n"
         }
         if let token: String = bpe.vocabulary.token(forId: Int(id)) {
-            let decodedToken = BytePairEncoder.decode(token: token)
+            let decodedToken = try BytePairEncoder.decode(token: token)
```
Comment (compnerd) [inline Models/Text/GPT2/GPT2.swift:163]: Should this be `try!` or doesn't `decodedToken` become optional?

File: Support/Text/BytePairEncoder.swift [hunk 2]
```diff
@@ -15,6 +15,10 @@
 import Foundation
 
 public struct BytePairEncoder {
+    public enum BPEError: Error {
+        case unsupported
```
Comment (compnerd) [inline Support/Text/BytePairEncoder.swift:19]: I think that `invalidEncoding` is better than `unsupported` ... it is supported, just cannot be encoded in a single byte.

Comment (texasmichelle) [reply Support/Text/BytePairEncoder.swift]: Changed to `invalidGeneratedToken`, to reflect that it's an intermediate encoded value generated by the bpe.

File: Models/Text/GPT2/GPT2.swift (hunk 1 above)
Comment (texasmichelle) [reply Models/Text/GPT2/GPT2.swift:163]: `decode()` does not return an optional and if it fails we just want to propagate the error, so neither `try!` nor `try?` make sense here.

**Pull request merged**